extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java

```diff
@@ -68,37 +73,54 @@
         Map<String, String> properties = new HashMap<>();
 
-        properties.put(MONGO_DB_CONNECTION_STRING, getConnectionString(matchingByType.get()));
+        setConnectionString(binding, properties);
+        setUsername(binding, properties);
+        setPassword(binding, properties);
 
         return Optional.of(new ServiceBindingConfigSource(BINDING_CONFIG_SOURCE_NAME, properties));
     }
 
-    private String getConnectionString(ServiceBinding binding) {
-        String user = getDbProperty(binding, DB_USER);
-        String password = getDbProperty(binding, DB_PASSWORD);
-        String hostPort = getHostPort(binding);
+    private void setConnectionString(ServiceBinding binding, Map<String, String> properties) {
+        String hostPort = getHost(binding);
+        String prefix = isSrv(binding) ? DB_PREFIX_SRV : DB_PREFIX_STANDARD;
         String database = getDbProperty(binding, DB_DATABASE);
-        String prefix = isPortProvided(hostPort) ? DB_PREFIX_STANDARD : DB_PREFIX_SRV;
-
+        String options = getDbProperty(binding, DB_OPTIONS);
+        boolean isOptionsNotBlank = isNotBlank(options);
         String connectionString;
-        if (isBlank(user)) {
-            if (isBlank(database)) {
-                connectionString = format(CONNECTION_STRING_WITHOUT_USER_NO_DB, prefix, hostPort);
-            } else {
-                connectionString = format(CONNECTION_STRING_WITHOUT_USER, prefix, hostPort, database);
+
+        if (isBlank(database)) {
+            connectionString = format(CONNECTION_STRING_WITH_ONLY_HOST, prefix, hostPort);
+
+            if (isOptionsNotBlank) {
+                // We need a trailing slash before options otherwise Mongo throws "contains options without trailing slash" error
+                // If the database value is not present, then we haven't yet added the trailing slash (and hence add it here)
+                connectionString += "/";
             }
         } else {
-            if (isBlank(database)) {
-                connectionString = format(CONNECTION_STRING_WITH_USER_NO_DB, prefix, user, password, hostPort);
-            } else {
-                connectionString = format(CONNECTION_STRING_WITH_USER, prefix, user, password, hostPort, database);
-            }
+            connectionString = format(CONNECTION_STRING_WITH_HOST_N_DB, prefix, hostPort, database);
+        }
+
+        if (isOptionsNotBlank) {
+            connectionString += "?" + options;
         }
 
         LOGGER.info(format("MongoDB connection string: [%s]", connectionString));
-        return connectionString;
+
+        properties.put(MONGO_DB_CONNECTION_STRING, connectionString);
+    }
+
+    private void setUsername(ServiceBinding binding, Map<String, String> properties) {
+        String username = getDbProperty(binding, DB_USER); // encodeIfContainsSpecialCharacters(getDbProperty(binding, DB_USER));
+        LOGGER.info(format("MongoDB username=%s", username));
```

Comment:
Are you sure these need to be INFO?

Reply:
Not really. They can be at the `debug` level too. I wasn't sure about the convention, hence, I added them at the `info` level. If that's not desirable then I can change it to `debug`.

Reply:
Yeah, let's make them debug please

Reply:
Let me also look at the integration tests as well. Could be because the connection string property doesn't contain username or password anymore.